### ISSUE TITLE
Unused private var warning is detected for used private var

### ISSUE THREAD

**Issue by nickpell** [borkdude/clj-kondo#915]
**version**

clj-kondo v2020.06.12

**platform**

Native

**problem**

```Unused private var``` warning is detected when the private variable is actually used.

**repro**

```
echo '(defn- a [] 1) (let [{:keys [a] :or {a a}} {}] (a))' | clj-kondo --lint -
<stdin>:1:8: warning: Unused private var user/a
linting took 34ms, errors: 0, warnings: 1
```

For clarity, the code with proper indentation is:
```
(defn- a [] 1)
(let [{:keys [a]
       :or {a a}} {}]
  (a))
```

**config**

No config.

**expected behavior**

Linting should pass without warnings.

**Comment by borkdude** [comment]
@nickpell I think

```
(let [{:keys [a] :or {a a}} {}] (a))
```

falls into the same category as:

```
(let [{:keys [a b] :or {b (inc a)}} {:a 1}] b)
```
where `b` refers to a previously destructured binding.

See discussion here: https://github.com/borkdude/clj-kondo/issues/782

One could argue that if one uses a default name equal to the binding, the code will guarantee that it's not one of the previous bindings, but one really has to dig into the code to see it, it's not very obvious.

The answer to this should probably be: don't do this.

I'm willing to support it if you can provide me with an authoritative answer from one of the Clojure core team members that this is documented behavior.

**Comment by puredanger** [comment]
Behavior of :or expressions that rely on bindings defined in the destructuring is undefined. No semantics has been established for this case, and you should not rely on whatever implementation details happen to impact current behavior. ie, don't do that.

**Comment by borkdude** [comment]
Thanks for chiming in @puredanger! Will support this.
